apps/foundation/pages/blog/[slug].tsx

```diff
@@ -1,17 +1,124 @@
-import React from 'react'
+import { FC } from 'react'
 
-import { useRouter } from 'next/router'
+import { dehydrate, QueryClient, QueryKey } from '@tanstack/react-query'
+import { API_URL, SITE_URL } from '@wsvvrijheid/config'
+import { Blog } from '@wsvvrijheid/types'
+import { BlogDetailTemplate } from '@wsvvrijheid/ui'
+import { getAuthorBlogs, getBlogBySlug, getBlogPaths } from '@wsvvrijheid/utils'
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations'
+import { MDXRemoteSerializeResult } from 'next-mdx-remote'
+import { serialize } from 'next-mdx-remote/serialize'
+import { NextSeoProps } from 'next-seo'
 
-const BlogInfo = () => {
-  const {
-    locale,
-    query: { slug },
-  } = useRouter()
+import { Layout } from '../../components'
+import i18nConfig from '../../next-i18next.config'
+
+type BlogPageProps = {
+  seo: NextSeoProps
+  queryKey: QueryKey
+  source: MDXRemoteSerializeResult
+  authorBlogs: Blog[]
+}
+
+const BlogDetailPage: FC<BlogPageProps> = ({
+  seo,
+  queryKey,
+  authorBlogs,
+  source,
+}) => {
+  // const { user } = useAuth()
   return (
-    <div>
-      {slug}-{locale}
-    </div>
+    <Layout seo={seo}>
+      <BlogDetailTemplate
+        auth={null}
+        queryKey={queryKey}
+        authorBlogs={authorBlogs}
+        source={source}
+      />
+    </Layout>
   )
 }
 
-export default BlogInfo
+export default BlogDetailPage
+
+export const getStaticPaths = async context => {
+  const paths = await getBlogPaths(context.locales)
+
+  return {
+    paths,
+    fallback: true,
+  }
+}
+
+export const getStaticProps = async context => {
+  const queryClient = new QueryClient()
+  const locale = context.locale
+
+  const slug = context.params?.slug
+
+  await queryClient.prefetchQuery({
+    queryKey: ['blog', locale, slug],
+    queryFn: () => getBlogBySlug(locale, slug),
+  })
+
+  const blog = queryClient.getQueryData<Blog>(['blog', locale, slug])
+
+  if (!blog) return { notFound: true }
+
+  const slugs =
+    blog.localizations?.reduce((acc, l) => {
+      acc[l.locale] = l.slug
+      return acc
+    }, {}) || {}
+
+  const title = blog?.title || null
+  const description = blog?.description || null
+  const adminUrl = API_URL
+  const siteUrl = SITE_URL
+  const image = blog.image
+  const url = `${siteUrl}/${locale}/blog/${slugs[locale]}`
```

Comment:
I think slugs doesn't contain the current blog's slug but it's localizations. We use slugs from pageProps when we switch the language.

```suggestion
  const url = `${siteUrl}/${locale}/blog/${blog.slug}`
```